tests/rptest/services/databricks_workspace.py

```diff
@@ -62,6 +63,10 @@ def create_catalog(self, bucket: str) -> "DatabricksCatalogInfo":
 
         self._location_names.add(bucket)
 
+        # Clean up any previous resources if reusing bucket name
+        self._cleanup_catalogs()
```

Comment:
This does not have any effect. `_cleanup_*` methods work on resources which are tracked by the service (see self._catalog_names for example). They can't cleanup anything which was not created by this service. This is on purpose. I.e. to avoid stepping over other concurrently running tests.

This periodic cleanup must be done outside of ducktape imho. The service/each test shouldn't spend time inspecting entire account to figure out what to clean up. Waste of time.